src/core/websocket/ws.ts

```
@@ -358,32 +394,53 @@ export class ReconnectingWebSocket {
             WebSocket = getGlobalWebSocket(),
         } = this._options;
 
+        // WebSocket implementation check
+        if (!isWebSocket(WebSocket)) {
```

Maybe this should throw an exception in the constructor? imho if we can't get a valid websocket implementation you should not even be able to construct a member of this class / enter into the connection loop.

---

That's a great point. This is also a pretty deep edge case that would mean they are not operating in the browser or in a node environment without `ws`. I like the idea of moving to the constructor.